### ISSUE TITLE
Disscuss "rules" of the game.

### ISSUE THREAD

**Issue by zenhack** [zenhack/iron-blogger2#8]
There's been some talk of changing how things work -- in particular, @fiendish is of the opinion that the punting mechanic provides the wrong incentive structure. I tend to agree. We should get a discussion going about what this should look like, though in the meantime we'll plow ahead implementing the old version.

//cc @paultag

**Comment by paultag** [comment]
Interesting. I'd love to hear more about this

**Comment by zenhack** [comment]
The basic problem that was raised was that once you've punted, there's not really much of an incentive to keep going. It would be nice to avoid that.

**Comment by fiendish** [comment]
Right. Things I've noticed about the debt structure...

1. After a user hits the punt limit, primary impetus for participation seems to be lost. The debt stops increasing as long as you don't pay it off. Paying it off means buying the ability to incur more debt. I know the system is supposed to be motivated by a desire to post more, but that is a perverse incentive.

2. A post per week may or may not be an apt schedule rigidity, depending on the blog. It should be possible to preserve the average while granting flexibility with an N per M schedule within reasonable ranges (perhaps we start with the current 1 per 1 week schedule). This is simply generalizing the schedule to be parametrized instead of fixed. Not a huge deal.

---

I believe a stable mechanism for dealing with both of these would be as follows:

* User has N empty post slots at the beginning of each M week period.
* At the end of the period, empty slots get converted to debt.
* If user posts on blog and has any empty slots, one empty slot is filled.
* If user posts on blog and does not have empty slots, debt is reduced by some TBD amount in the range [0-slot_penalty).

**Comment by zenhack** [comment]
I like it. I'm tempted to suggest the credit for "late" posts be a function of how old the empty slots are -- this gives a nice incentive to get last week's post in this week, rather than just saying "eh, I'll do an extra one someday." Maybe an exponential backoff? Thoughts?

**Comment by fiendish** [comment]
Hmm. Maybe. If TBD reduction amount is < slot_penalty, then you have to post multiple times to clear debt even without a sliding amount. Perhaps if it starts at ~slot_penalty and gradually decreases then getting a post in a day late is better than a week late.

**Comment by zenhack** [comment]
So here's what I'm thinking of implementing:

For now, I'm not going to worry about variable period lengths, just assume the basic 1 post per week.

* A post which is missed entirely costs the user $5
* A late post costs the user $N where N is the number of weeks (rounded up) since the due date, up to a maximum of $5.
* The first post written during a given period counts towards *that* period's deadline, regardless of late posts. Subsequent posts in the same week each count towards the most recent missed deadline. 

@paultag, @fiendish, thoughts?

**Comment by fiendish** [comment]
First bullet is redundant, covered by second.
"posts" vs "most recent missed deadline" in the third bullet implies that "missed deadline" is popped from a stack like shown. Correct?

| week 1 |  week 2 |  week 3 |  week 4 |  week 5 | week 6 | Posts in Week 7 | owed |
|---|---|---|---|---|---|---|---|
| :x: $5 | :x: $5 | :x: $4 | :x: $3 | :x: $2 | :x: $1 | 0 | $20 |
| :x: $5 | :x: $5 | :x: $4 | :x: $3 | :x: $2 | :white_check_mark: | 1 |$19 |
| :x: $5 | :x: $5 | :x: $4 | :x: $3 | :white_check_mark: | :white_check_mark: | 2 |$17 |
| :x: $5 | :x: $5 | :x: $4 | :white_check_mark:  | :white_check_mark: | :white_check_mark: | 3 |$14 |
| :x: $5 | :x: $5 | :white_check_mark: | :white_check_mark:  | :white_check_mark: | :white_check_mark: | 4 |$10 |
| :x: $5 | :white_check_mark: | :white_check_mark: | :white_check_mark:  | :white_check_mark: | :white_check_mark: | 5 |$5 |
|  :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark:  | :white_check_mark: | :white_check_mark: | 6 |$0 |

**Comment by zenhack** [comment]
(Had this conversation in person, but to have it on the record...)

> First bullet is redundant, covered by second.

By "missed entirely," I meant "never turned in," while by "late post," I meant one which has been turned in, but was late.

> "posts" vs "most recent missed deadline" in the third bullet implies that "missed deadline" is popped from a stack like shown. Correct?

Perhaps this was a litle confusing. Let me try again:

* A post which is on-time incurs no debt.
* A post which is turned in N weeks (rounded up) late incurs a debt of $N, up to a maximum of $5.
* A post which is never turned in incurs a debt of $5.
* When a post is turned in,
  * If it is the first post of the week, it is counted as that weeks post.
  * Otherwise, it is counted as being a "late" post for the most recent week with no posts.

To correct the chart as well (Assume user has not posted since before week 1):

| week 1 |  week 2 |  week 3 |  week 4 |  week 5 | week 6 | week 7 (current) | Posts in Week 7 | owed |
|---|---|---|---|---|---|---|---|---|
| :x: $5 | :x: $5 | :x: $5 | :x: $5 | :x: $5 | :x: $5 | :x: $5 | 0 | $35 |
| :x: $5 | :x: $5 | :x: $5 | :x: $5 | :x: $5 | :x: $5 | :white_check_mark: $0 | 1 |$30 |
| :x: $5 | :x: $5 | :x: $5 | :x: $5 | :x: $5 | :white_check_mark: $1 | :white_check_mark: $0 | 2 |$26 |
| :x: $5 | :x: $5 | :x: $5 | :x: $5 | :white_check_mark: $2 | :white_check_mark: $1 | :white_check_mark: $0 | 3 |$23 |
| :x: $5 | :x: $5 | :x: $5 | :white_check_mark: $3 | :white_check_mark: $2 | :white_check_mark: $1 | :white_check_mark: $0 | 4 |$21 |
| :x: $5 | :x: $5 | :white_check_mark: $4 | :white_check_mark: $3 | :white_check_mark: $2 | :white_check_mark: $1 | :white_check_mark: $0 | 5 |$20 |
| :x: $5 |  :white_check_mark: $5 | :white_check_mark: $4 | :white_check_mark: $3 | :white_check_mark: $2 | :white_check_mark: $1 | :white_check_mark: $0 | 6 |$20 |

**Comment by zenhack** [comment]
It's about time we got the rest of the participants to weigh in on this --- would have been ideal to do so *before* I went and built it, but oh well. Queue at-mentioning of everyone whose github username I can find. If you notice I've missed someone, add them. Or suggest a better venue to have this discussion (we definitely have some folks not on github).

//cc @mjg59 @fiendish @maalox @mollydb @madprime @paultag @shaunagm @jywarren

Would appreciate your thoughts.

**Comment by shaunagm** [comment]
I like this new structure.  It's a little complicated, but anyone who thinks it's too complicated can just write a post every week - not paying attention to the rules for decreasing debt doesn't cost any more than the $5 per missed post from before.

**Comment by zenhack** [comment]
There was some additional discussion via email, and we settled on this plus a debt cap of $30 between parties (see #32). I'm going to close this; we've got a set of rules for the time being.
